web/src/message_fetch.js

```diff
@@ -48,7 +48,10 @@ function process_result(data, opts) {
         all_messages_data.add_messages(messages);
     }
 
-    if (messages.length !== 0) {
+    if (messages.length !== 0 && opts.msg_list) {
+        // This was a load_messages call to only populate the data,
+        // which is used outside of the message_list. So, the caller
+        // should be responsible for adding the messages.
         message_util.add_old_messages(messages, opts.msg_list);
     }
```

Comment:
Hmm, I feel like the thing we'd expect to have happen in an else clause here is `msg_list_data.add_messages(messages)`

Reply:
Makes sense:

```js
    if (messages.length !== 0) {
        if (opts.msg_list) {
            // Since this adds messages to the MessageList and renders MessageListView,
            // we don't need to call it if msg_list was not defined by the caller.
            message_util.add_old_messages(messages, opts.msg_list);
        } else {
            opts.msg_list_data.add_messages(messages);
        }
    }

```